sql/src/main/java/io/crate/operation/reference/sys/node/SysNodesExpressionFactories.java

```diff
@@ -0,0 +1,442 @@
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.node;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.RowCollectExpression;
+import io.crate.metadata.RowContextCollectorExpression;
+import io.crate.metadata.expressions.RowCollectExpressionFactory;
+import io.crate.metadata.sys.SysNodesTableInfo;
+import io.crate.monitor.ExtendedFsStats;
+import io.crate.monitor.ThreadPools;
+import io.crate.operation.reference.sys.node.fs.NodeFsDataExpression;
+import io.crate.operation.reference.sys.node.fs.NodeFsDisksExpression;
+import io.crate.operation.reference.sys.node.fs.NodeFsExpression;
+import io.crate.operation.reference.sys.node.fs.NodeFsTotalExpression;
+import org.apache.lucene.util.BytesRef;
+
+import java.util.Map;
+
+public class SysNodesExpressionFactories {
+
+    public SysNodesExpressionFactories() {
+    }
+
+    public static Map<ColumnIdent, RowCollectExpressionFactory> getSysNodesTableInfoFactories() {
+        return ImmutableMap.<ColumnIdent, RowCollectExpressionFactory>builder()
+            .put(SysNodesTableInfo.Columns.ID, new RowCollectExpressionFactory() {
+                @Override
+                public RowCollectExpression create() {
+                    return new RowContextCollectorExpression<DiscoveryNodeContext, BytesRef>() {
+                        @Override
+                        public BytesRef value() {
+                            return row == null ? null : row.id();
+                        }
+                    };
+                }
+            })
+            .put(SysNodesTableInfo.Columns.NAME, new RowCollectExpressionFactory() {
+                @Override
+                public RowCollectExpression create() {
+                    return new RowContextCollectorExpression<DiscoveryNodeContext, BytesRef>() {
+                        @Override
+                        public BytesRef value() {
+                            return row == null ? null : row.name();
```

Comment:
why are there null checks for the first two expressions but not in the other expressions?